app/views/render_async/_request_vanilla.js.erb

```diff
@@ -52,5 +58,14 @@
 
     var body = "<%= escape_javascript(data.to_s.html_safe) %>";
     request.send(body);
+  };
+
+  <% if turbolinks %>
+  document.addEventListener("turbolinks:load", function (e) {
+    e.target.removeEventListener(e.type, arguments.callee);
```

Comment:
Hey @eclemens, can you maybe explain a bit why are we removing the listener here? :)

Reply:
With Turbolinks, window.document is the same for all pages of the application/website. Therefore, if you add an event to it and do not delete it after it has been called, the events will start to accumulate and all of them will be executed every time you change pages. This is not a problem without Turbolinks since the document.onload is only called once per page and all events are discarded between each page change.

Also if you look at the implementation of jQuery I used `jQuery.one()` instead of `jQuery.on()` to produce the same behaviour.

Translated with www.DeepL.com/Translator

Reply:
Great, thanks for the explanation :)